[v8.12] 8.12.2 release (#13607)
coq/coq: doc/sphinx/changes.rst

  fixes `#12332 <https://github.com/coq/coq/issues/12332>`_,
  by Théo Zimmermann and Jim Fehrle).

Changes in 8.12.2
~~~~~~~~~~~~~~~~~

**Notations**

- **Fixed:**

>   8.12 regression causing notations mentioning a coercion not being used

Reviewer: ```suggestion
  8.12 regression causing notations mentioning a coercion to be ignored
```

or "to not be used"

  (`#13436 <https://github.com/coq/coq/pull/13436>`_,
  fixes `#13432 <https://github.com/coq/coq/issues/13432>`_,
  by Hugo Herbelin).

**Tactics**

- **Fixed:**

>   8.12 regression in the inference of implicit arguments in :tacn:`exists`

Reviewer: ```suggestion
  8.12 regression: incomplete inference of implicit arguments in :tacn:`exists`
```
